Add support for scope transformers

IamBlueSlime opened the pull request:
Currently, the extra information sent to Sentry are static and defined on the interceptor initialization. Some may want to add some dynamic information before sending the exception to Sentry.

One simple example, I use Elastic APM (https://www.elastic.co/guide/en/apm/get-started/current/overview.html) to monitor my Nest applications and I want to inject the request id to the Sentry message to trace the original request made by the client.

This is simply implemented by providing an array of functions accepting the Sentry's scope as parameter.
Added a test case according to my modifications.

IamBlueSlime commented:
(The 14.x test suite is failing due to a random timeout error related to websockets, should be safe to merge as-it.)

mentos1386 commented:
@IamBlueSlime thanks for making this PR!

Can this be used with global interceptor?

Instead of adding `transformers` to `Interceptor`. Could we add a decorator `ReavenTransformer` that could be used "anywhere" to add additional data.
```ts
@RavenTransformer((scope: Scope) => { scope.addExtra('important key', 'useful value'))
```

Related with https://github.com/mentos1386/nest-raven/issues/22

IamBlueSlime commented:
This could be used with global interceptor in the same way as local ones. Adding a `RavenTransformer` decorator is not a bad idea, I'll try to implement it today. Thanks for the suggestion!

IamBlueSlime commented:
@mentos1386 I'm done with the implementation of the `RavenTransformer` decorator.
I've refactored a little the type-specific methods to not be too verbose, let me know if you want me to revert my changes.
I've also added some more test cases to cover my new adds.

mentos1386 commented:
@IamBlueSlime awesome!

One thing that still worries me, is how useful is the transformer? Should we pass `ExecutionContext` besides the `scope`?

I haven't worked with NestJS for a while, so i might be thinking about this wrong. Can you provide a small example on how you would provide "request id" with `RavenTransformer` decorator?

IamBlueSlime commented:
@mentos1386 Here is the implementation I want to use in my Nest application:

![image](https://user-images.githubusercontent.com/6763873/90637136-e6179e00-e22b-11ea-9fb4-0a387c285271.png)

As you can see, my usage is pretty simple but I've managed to be as most open-minded as possible with the implementation of the transformer. Let me know if you want me to add more data to the function.
